### PR TITLE
Use esm exports from react-use library

### PR DESCRIPTION
**Pull request by szubster**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .github/vale/config/vocabularies/Backstage/accept.txt [hunk 1]
```diff
@@ -114,6 +114,7 @@ env
 Env
 esbuild
 eslint
+esm
```
Comment (freben) [inline .github/vale/config/vocabularies/Backstage/accept.txt:117]: probably better to make it ESM uppercase in the changeset?

**Review by freben — COMMENTED** [review]
What's the reason for this change?

**Comment by szubster** [comment]
It saves some bytes in the bundle (not much, but still it's "free").
ESM is tree-shakeable, and CJS in `lib` is not.

File: packages/cli/config/jest.js [hunk 2]
```diff
@@ -43,7 +43,6 @@ const transformIgnorePattern = [
   'highlight\\.js',
   'prismjs',
   'json-schema',
-  'react-use',
```
Comment (freben) [inline packages/cli/config/jest.js:46]: Hm does this mean that if someone out there DOES still use `/lib/`, they are screwed? Remember, this CLI is shipped and used worldwide. Or what's the impact of this line?

Comment (freben) [reply packages/cli/config/jest.js:46]: This needs a changeset too BTW. And if we do keep the change, the changeset should probably very loudly say that users of `react-use` should switch the way that they do their imports.

**Review by freben — COMMENTED** [review]
needs a rebase + i have a question

File: packages/cli/config/jest.js (hunk 2 above)
Comment (szubster) [reply packages/cli/config/jest.js:46]: Rebased.
I have also added changeset to cli, and tried to minimize disruption in tests, by leaving `react-use/lib` in the list.
This way both `react-use/lib` and `react-use/esm` work in tests.

File: .changeset/wild-walls-sparkle.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/cli': patch
+---
+
+Ignore transforming only on `react-use/lib`, not whole `react-use` in jest.
+
+** POTENTIAL BREAKAGE **
+If you tests fail, please change to use path import from `react-use/esm/`. It is also recommended to migrate from `react-user/lib` imports to `react-use/esm`
```
Comment (awanlin) [inline .changeset/wild-walls-sparkle.md:8]:
```suggestion
If your tests fail, please change to use path import from `react-use/esm/`. It is also recommended to migrate from `react-user/lib` imports to `react-use/esm`
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @szubster, looks good for the Azure DevOps, DevTools, and Linguist plugins 🚀 

There was a small typo in the changeset and it looks like some conflicts have crept in so please rebase 👍

**Comment by szubster** [comment]
> Thanks for this @szubster, looks good for the Azure DevOps, DevTools, and Linguist plugins 🚀
> 
> There was a small typo in the changeset and it looks like some conflicts have crept in so please rebase 👍

done

**Comment by freben** [comment]
Off we go then!

**Pull request merged**